0012-unicode-strings.md

```diff
@@ -55,6 +54,23 @@ The default prior to Dafny 4.0 is `/unicodeChar:0`, but the default as of Dafny
 This change in definition can cause previously verified code to no longer verify,
 since expressions such as `someIntValue as char` or `someCharValue + 'a'` are no longer allowed to result in
 surrogate code points.
+Using `/unicodeChar:1` will also cause the parser to reject invalid string literals,
+such as the `"\uD800"` value above.
+This ensures Dafny string literals can always be represented in UTF-8, UTF-16, or any other Unicode encoding.
+
+The exact representation of strings at runtime, including the particular encoding,
+is an implementation detail of a particular backend, and will often be optimized for the idioms and support
+of the target environment. 
+This also applies to string literals in Dafny source code:
```

Comment:
This is better but still ambiguous.  Maybe it's enough to say that strings literals are as in Dafny 3, except for the fact that unpaired surrogates are disallowed?

Reply:
That's only true w.r.t. what literals are permitted in source text though (with the added `\U{X..X}` escape sequence), not how they are represented at runtime. I do want to say that, unlike in Dafny 3, users shouldn't assume a string literal becomes a sequence where every ASCII character or `\u` escape sequence literally becomes an individual element of that sequence. I want to call out that if you compile your program to Go, for example, your literal will likely become a Go `string` value wrapped as a `Dafny.Sequence<rune>`.

Reply:
Again I'm hoping that dropping support for `\u` makes this clearer.